### PR TITLE
Zfs fs send

### PR DESCRIPTION
**Pull request by tych0**

### CODE REVIEW

**Comment by stgraber** [comment]
Looks like you're running into some zfs race conditions :)

**Comment by tych0** [comment]
Urgh. More goofy code here we come.

**Comment by tych0** [comment]
jenkins: test this please

**Comment by tych0** [comment]
jenkins: test this please

**Comment by tych0** [comment]
jenkins: test this please

**Comment by tych0** [comment]
ok, let me rebase this into something reasonable and it looks like it's good to merge.

**Comment by tych0** [comment]
Ok, I think this version should be good to go once jenkins agrees.

File: test/suites/migration.sh [hunk 1]
```diff
@@ -34,12 +29,8 @@ test_migration() {
...
   fi
 
-  # FIXME: make this backend agnostic
-  if [ "${LXD_BACKEND}" = "dir" ]; then
-    # since we didn't snapshot above, we can't do this here
-    lxc copy l1:nonlive2/snap0 l2:nonlive3
-    [ -d "${LXD2_DIR}/containers/nonlive3/rootfs/bin" ]
-  fi
+  lxc copy l1:nonlive2/snap0 l2:nonlive3
+  [ -d "${LXD2_DIR}/containers/nonlive3/rootfs/bin" ]
```
Comment (stgraber) [inline test/suites/migration.sh]: That line will fail with lvm as rootfs isn't mounted unless the container is running.

**Comment by tych0** [comment]
Passed once, let's try again,

jenkins: test this please

**Comment by stgraber** [comment]
The ZFS bits passed again, sorry for the rest of the breakage, that was me adding lvm support to the builders.

Second of 3 tries.

jenkins: test this please

**Comment by stgraber** [comment]
zfs looks good again.

Third of 3 tries.

jenkins: test this please

**Comment by stgraber** [comment]
All good, only lvm is failing.

**Pull request merged**